Bitcoin Core `configure` fails after #1084 was merged

Bitcoin Core's secp subtree is currently at `44c2452fd387f7ca604ab42d73746e7d3a44d8a2`. Updating the subtree to `cd470333351bd6d90296352f2a957f38bbdaf014` (merging #1084) causes Bitcoin Core CI failure as it cannot run the config scripts anymore as seen [here](https://github.com/dhruv/bitcoin-core-ci/pull/14/checks)

I don't know much about build systems, but am happy to learn and help with guidance.

> Hmmm.... So apparently Core's CI sets some explicit cache file that caches some configuration values:

Hm no, so this done automatically by autoconf when running `AC_CONFIG_SUBDIRS`. And IIUC then it complains that the environment has changed in between the two configure runs (core and secp256k1), namely
```
configure: error: `PKG_CONFIG_PATH' was not set in the previous run
```

@dhruv Can you test https://github.com/bitcoin-core/secp256k1/pull/1090? I suspect that to be the issue instead of #1084 but  I could be wrong.

--- a/configure.ac
+++ b/configure.ac
@@ -43,8 +43,6 @@ AS_UNSET(ac_cv_prog_AR)
 AS_UNSET(ac_cv_prog_ac_ct_AR)
 LT_INIT([win32-dll])
 
-PKG_PROG_PKG_CONFIG
-
 build_windows=no
 
 case $host_os in

